src/renderer/actions/register.ts

```diff
@@ -65,32 +65,40 @@ export interface IWithdrawalKeyAction extends Action<RegisterActionTypes> {
 
 // After password action
 export const afterPasswordAction = (password: string) => {
-    return (dispatch: Dispatch<Action<RegisterActionTypes>>, getState: () => IRootState): void => {
+    return async (dispatch: Dispatch<Action<RegisterActionTypes>>, getState: () => IRootState): Promise<void> => {
         // 1. Save to keystore
-        const signingKey = getState().register.signingKey;
-        V4Keystore.create(
-            `${KEYSTORE_DEFAULT_DIRECTORY}/${getV4Filename()}.json`, 
-            password, new Keypair(PrivateKey.fromHexString(signingKey))
+        dispatch(startRegistrationSubmission());
+        const signingKey = PrivateKey.fromBytes(
+            Buffer.from(getState().register.signingKey.slice(2), "hex")
+        );
+        const accountDirectory = path.join(getConfig().storage.accountsDir, "default");
+        await V4Keystore.create(
+            path.join(accountDirectory, PublicKey.fromPrivateKey(signingKey).toHexString() + ".json"),
+            password, new Keypair(signingKey)
         );
 
         // 2. Save account to db
         const account = new CGAccount({
-            name: "Test Account",
-            directory: KEYSTORE_DEFAULT_DIRECTORY,
+            name: "Default",
```

Comment:
also DEFAULT_ACCOUNT ?

Reply:
We don't use this field, it will be used later on if we introduce multi-account stuff